solarforecastarbiter/datamodel.py

```diff
@@ -1230,10 +1233,72 @@ def __post_init__(self):
                 'One of observation_id OR aggregate_id must be set')
 
 
+def __check_plot_spec__(plot_spec):
+    """Ensure that the provided plot specification is a valid JSON object"""
+    try:
+        spec_dict = json.loads(plot_spec)
+        validate(instance=spec_dict, schema={'type': 'object'})
+    except (json.JSONDecodeError, ValidationError):
+        raise ValueError('Figure spec must be a valid json object.')
+
+
 @dataclass(frozen=True)
 class ReportFigure(BaseModel):
+    """Parent class for different types of Report Figures"""
+    def __post_init__(self):
+        if type(self) == ReportFigure:
+            raise ValueError("Invalid Report Figure. Figures must be of class "
+                             "PlotlyReportFigure or BokehReportFigure.")
+
+    @classmethod
+    def from_dict(model, input_dict, raise_on_extra=False):
+        dict_ = input_dict.copy()
+        if model != ReportFigure:
+            return super().from_dict(dict_, raise_on_extra)
+        figure_class = dict_.get('figure_class')
+        if figure_class == 'plotly':
+            return PlotlyReportFigure.from_dict(dict_, raise_on_extra)
+        elif figure_class == 'Bokeh':
+            return BokehReportFigure.from_dict(dict_, raise_on_extra)
+        else:
+            raise NotImplementedError(
+                f'Do not know how to process {dict_} into a ReportFigure.')
+
+
+@dataclass(frozen=True)
+class PlotlyReportFigure(ReportFigure):
     """A class for storing metric plots for a report with associated metadata.
 
+    Parameters
+    ----------
+    name: str
+        A descriptive name for the figure.
+    spec: str
+        JSON string representation of the plotly plot.
+    svg: str
```

Comment:
what would we run into if we wanted this to default to none or empty string and have a pdf field?

Reply:
I don't think it would be much of an issue, we'd just have to add the pdf creation to the RawReport computation, and add templating logic. If there's some gotcha we haven't considered, I don't think it can be too much of an issue since we aren't actively utilizing the svgs for anything. We'd just have to recompute reports so their figures conform to the datamodel.

Reply:
Ok, let's add a pdf parameter that is optional and make an ``output_pdf`` function for the plotly figures

Reply:
or actually, ok to just make an issue and we can do it later